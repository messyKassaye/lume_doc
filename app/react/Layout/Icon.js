import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';

import { Icon as UIIcon, CountryList } from 'UI';

function getFaSize(size) {
  switch (size) {
    case 'xs':
      return 'sm';
    case 'sm':
      return 'lg';
    case 'md':
      return '2x';
    case 'lg':
      return '2x';
    case 'xlg':
      return '2x';
    default:
      return 'sm';
  }
}

export class Icon extends Component {
  getJSData() {
    const { data } = this.props;
    let _data = data;

    if (data && _data.toJS) {
      _data = _data.toJS();
    }

    return _data;
  }

  getIcon(_data) {
    const { className, size } = this.props;
    let icon;

    if (_data.type === 'Icons') {
      icon = <UIIcon icon={`${_data._id}`} size={`${getFaSize(size)}`} />;
    }

    if (_data.type === 'Flags') {
      const flagClassName = `flag-icon flag-icon-${CountryList.get(_data._id).cca2}`.toLowerCase();
      const componentClassName = `${className} ${flagClassName}`;
      icon = <span className={componentClassName} />;
    }

    return icon;
  }

  render() {
    const { className } = this.props;
    let html = null;
    const _data = this.getJSData();

    if (_data && _data._id) {
      const icon = this.getIcon(_data);
      html = <span className={className}>{icon}</span>;
    }

    return html;
  }
}

Icon.defaultProps = {
  data: undefined,
  className: undefined,
  size: undefined,
};

Icon.propTypes = {
  data: PropTypes.instanceOf(Object),
  className: PropTypes.string,
  size: PropTypes.string,
};

export default connect()(Icon);
