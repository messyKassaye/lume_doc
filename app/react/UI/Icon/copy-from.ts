/* eslint-disable max-len */

const copyFromSvgPath =
  'M11.5 3H2.5C1.6875 3 1 3.6875 1 4.5V13.5C1 14.3438 1.6875 15 2.5 15H11.5C12.3438 15 13 14.3438 13 13.5V12.1624H9.02806V13.5556C9.02806 14.1781 8.25736 14.5041 7.81274 14.0595L5.26353 11.4807C4.96711 11.2139 4.96711 10.7693 5.26353 10.5025L7.81274 7.92365C8.25736 7.47902 9.02806 7.80508 9.02806 8.42756V9.79109H13V4.5C13 3.6875 12.3438 3 11.5 3ZM14 9.79109V6.41033C14 6.27055 13.9883 6.13349 13.966 6.00006H19.5C20.3438 6.00006 21 6.68756 21 7.50006V16.5001C21 17.3438 20.3438 18.0001 19.5 18.0001H10.5C9.6875 18.0001 9 17.3438 9 16.5001V15.9488H11.5385C12.8979 15.9488 14 14.8467 14 13.4873V12.1624H16.6442C16.8517 12.1624 16.9999 12.0142 16.9999 11.8067V10.1468C16.9999 9.96894 16.8517 9.79109 16.6442 9.79109H14Z';
export const copyFrom = {
  prefix: 'fas',
  iconName: 'copy-from',
  icon: [16, 16, [], 'e001', copyFromSvgPath],
};
