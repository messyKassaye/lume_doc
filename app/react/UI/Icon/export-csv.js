/* eslint-disable max-len */

const exportCsvSvgPath =
  'M 13.3594 0.703125 L 17.2969 4.59375 C 17.7188 5.01562 18 5.625 18 6.1875 V 21.7969 C 18 23.0156 16.9688 24 15.7031 24 H 10.4819 V 21.7969 H 15.75 V 8.29688 H 10.8281 C 10.2188 8.29688 9.70312 7.78125 9.70312 7.17188 V 2.29688 H 2.20312 V 21.7969 H 6.40312 V 24 H 2.20312 C 0.984375 24 0 23.0156 0 21.7969 V 2.29688 C 0 1.03125 0.984375 0.046875 2.20312 0 H 11.8125 C 12.375 0 12.9375 0.28125 13.3594 0.703125 Z M 15.5625 6.04688 L 11.9531 2.4375 V 6.04688 H 15.5625 Z M 9.10817 19.9289 C 9.44807 19.9289 9.78797 20.211 9.78797 20.606 V 25.3263 H 12.5072 C 13.1303 25.3263 13.4136 26.0598 13.017 26.4548 L 8.93822 30.5173 C 8.65497 30.7994 8.14511 30.7994 7.86186 30.5173 L 3.78305 26.4548 C 3.3865 26.0598 3.66975 25.3263 4.2929 25.3263 H 7.06876 V 20.606 C 7.06876 20.211 7.35201 19.9289 7.74856 19.9289 H 9.10817 Z M 6.69785 14.9746 C 6.66894 15.6404 6.48495 16.1444 6.14588 16.4866 C 5.80681 16.8289 5.32843 17 4.71075 17 C 4.06153 17 3.56344 16.7834 3.21649 16.3503 C 2.87216 15.9144 2.7 15.2941 2.7 14.4893 V 13.5067 C 2.7 12.7045 2.87873 12.0869 3.2362 11.6537 C 3.59367 11.2179 4.09044 11 4.72652 11 C 5.35209 11 5.82652 11.1778 6.14982 11.5334 C 6.47575 11.889 6.66105 12.3997 6.70573 13.0655 H 5.54265 C 5.53214 12.6537 5.46906 12.3703 5.3534 12.2152 C 5.24038 12.0575 5.03142 11.9786 4.72652 11.9786 C 4.41637 11.9786 4.19689 12.0896 4.0681 12.3115 C 3.93931 12.5307 3.87097 12.893 3.86308 13.3984 V 14.5013 C 3.86308 15.0816 3.92616 15.4799 4.05233 15.6965 C 4.18112 15.9131 4.4006 16.0214 4.71075 16.0214 C 5.01565 16.0214 5.22593 15.9465 5.34158 15.7968 C 5.45723 15.6444 5.52294 15.3703 5.53871 14.9746 H 6.69785 Z M 9.67493 15.3877 C 9.67493 15.1497 9.61448 14.9706 9.49357 14.8503 C 9.37529 14.7273 9.15845 14.6003 8.84303 14.4693 C 8.26741 14.2473 7.85343 13.988 7.6011 13.6912 C 7.34877 13.3917 7.2226 13.0388 7.2226 12.6324 C 7.2226 12.1404 7.39345 11.746 7.73515 11.4492 C 8.07947 11.1497 8.51579 11 9.04411 11 C 9.39632 11 9.71042 11.0762 9.9864 11.2286 C 10.2624 11.3783 10.474 11.5909 10.6212 11.8663 C 10.771 12.1417 10.8459 12.4545 10.8459 12.8048 H 9.6907 C 9.6907 12.5321 9.63288 12.3249 9.51723 12.1832 C 9.40421 12.0388 9.23993 11.9666 9.0244 11.9666 C 8.82201 11.9666 8.6643 12.0281 8.55128 12.1511 C 8.43826 12.2714 8.38174 12.4345 8.38174 12.6404 C 8.38174 12.8008 8.44483 12.9465 8.57099 13.0775 C 8.69716 13.2059 8.92057 13.3396 9.24124 13.4786 C 9.8011 13.6845 10.2072 13.9372 10.4595 14.2366 C 10.7145 14.5361 10.842 14.9171 10.842 15.3797 C 10.842 15.8877 10.6829 16.2848 10.3649 16.5709 C 10.0469 16.857 9.61448 17 9.06777 17 C 8.69716 17 8.3594 16.9225 8.0545 16.7674 C 7.74961 16.6123 7.51042 16.3904 7.33694 16.1016 C 7.16609 15.8128 7.08067 15.4719 7.08067 15.0789 H 8.24375 C 8.24375 15.4158 8.30815 15.6604 8.43694 15.8128 C 8.56573 15.9652 8.77601 16.0414 9.06777 16.0414 C 9.47254 16.0414 9.67493 15.8235 9.67493 15.3877 Z M 14.0592 11 L 13.0932 15.2353 L 12.1312 11 H 10.8459 L 12.49 16.8396 H 13.6964 L 15.3524 11 H 14.0592 Z';
export const exportCsv = {
  prefix: 'fas',
  iconName: 'export-csv',
  icon: [22, 22, [], 'e001', exportCsvSvgPath],
};
