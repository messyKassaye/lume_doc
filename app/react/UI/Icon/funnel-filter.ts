/* eslint-disable max-len */

const funnelFilterSvgPath =
  'M4.86858 9.74499c-.18725 0-.3745.18725-.3745.37451v.749c0 .2107.18725.3745.3745.3745h2.26284c.18725 0 .3745-.1638.3745-.3745v-.749c0-.18726-.18725-.37451-.3745-.37451H4.86858zm-2.99605-5.9921c-.18725 0-.3745.18725-.3745.3745v.74901c0 .21066.18725.37451.3745.37451h8.25497c.1872 0 .3745-.16385.3745-.37451v-.74901c0-.18725-.1873-.3745-.3745-.3745H1.87253zm1.49803 2.99605c-.18726 0-.37451.18725-.37451.3745v.74902c0 .21065.18725.3745.37451.3745h5.25888c.18726 0 .37451-.16385.37451-.3745v-.74902c0-.18725-.18725-.3745-.37451-.3745H3.37056zM.374506.756836C.187253.756836 0 .944089 0 1.13134v.74901c0 .21066.187253.37451.374506.37451H11.6255c.1872 0 .3745-.16385.3745-.37451v-.74901c0-.187251-.1873-.374504-.3745-.374504H.374506z';
export const funnelFilter = {
  prefix: 'fas',
  iconName: 'funnel-filter',
  icon: [12, 12, [], 'e001', funnelFilterSvgPath],
};
