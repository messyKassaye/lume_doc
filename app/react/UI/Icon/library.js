import { library } from '@fortawesome/fontawesome-svg-core';
import { faAlignJustify } from '@fortawesome/free-solid-svg-icons/faAlignJustify';
import { faAngleLeft } from '@fortawesome/free-solid-svg-icons/faAngleLeft';
import { faAngleRight } from '@fortawesome/free-solid-svg-icons/faAngleRight';
import { faArrowLeft } from '@fortawesome/free-solid-svg-icons/faArrowLeft';
import { faArrowRight } from '@fortawesome/free-solid-svg-icons/faArrowRight';
import { faArrowUp } from '@fortawesome/free-solid-svg-icons/faArrowUp';
import { faBars } from '@fortawesome/free-solid-svg-icons/faBars';
import { faBook } from '@fortawesome/free-solid-svg-icons/faBook';
import { faBullhorn } from '@fortawesome/free-solid-svg-icons/faBullhorn';
import { faCalculator } from '@fortawesome/free-solid-svg-icons/faCalculator';
import { faCalendar } from '@fortawesome/free-solid-svg-icons/faCalendar';
import { faCaretDown } from '@fortawesome/free-solid-svg-icons/faCaretDown';
import { faCaretUp } from '@fortawesome/free-solid-svg-icons/faCaretUp';
import { faChartBar } from '@fortawesome/free-solid-svg-icons/faChartBar';
import { faCheck } from '@fortawesome/free-solid-svg-icons/faCheck';
import { faCheckCircle } from '@fortawesome/free-solid-svg-icons/faCheckCircle';
import { faChevronLeft } from '@fortawesome/free-solid-svg-icons/faChevronLeft';
import { faChevronRight } from '@fortawesome/free-solid-svg-icons/faChevronRight';
import { faCircle } from '@fortawesome/free-solid-svg-icons/faCircle';
import { faClock } from '@fortawesome/free-solid-svg-icons/faClock';
import { faClone } from '@fortawesome/free-solid-svg-icons/faClone';
import { faCloudUploadAlt } from '@fortawesome/free-solid-svg-icons/faCloudUploadAlt';
import { faCloudDownloadAlt } from '@fortawesome/free-solid-svg-icons/faCloudDownloadAlt';
import { faCode } from '@fortawesome/free-solid-svg-icons/faCode';
import { faCoffee } from '@fortawesome/free-solid-svg-icons/faCoffee';
import { faCog } from '@fortawesome/free-solid-svg-icons/faCog';
import { faComment } from '@fortawesome/free-solid-svg-icons/faComment';
import { faCubes } from '@fortawesome/free-solid-svg-icons/faCubes';
import { faCircle as faEmptyCircle } from '@fortawesome/free-regular-svg-icons/faCircle';
import { faEnvelope } from '@fortawesome/free-solid-svg-icons/faEnvelope';
import { faExchangeAlt } from '@fortawesome/free-solid-svg-icons/faExchangeAlt';
import { faExclamationTriangle } from '@fortawesome/free-solid-svg-icons/faExclamationTriangle';
import { faFile } from '@fortawesome/free-solid-svg-icons/faFile';
import { faFlask } from '@fortawesome/free-solid-svg-icons/faFlask';
import { faFilter } from '@fortawesome/free-solid-svg-icons/faFilter';
import { faFilePdf } from '@fortawesome/free-solid-svg-icons/faFilePdf';
import { faFileImage } from '@fortawesome/free-solid-svg-icons/faFileImage';
import { faFont } from '@fortawesome/free-solid-svg-icons/faFont';
import { faGavel } from '@fortawesome/free-solid-svg-icons/faGavel';
import { faHome } from '@fortawesome/free-solid-svg-icons/faHome';
import { faImage } from '@fortawesome/free-solid-svg-icons/faImage';
import { faInfoCircle } from '@fortawesome/free-solid-svg-icons/faInfoCircle';
import { faLanguage } from '@fortawesome/free-solid-svg-icons/faLanguage';
import { faLightbulb } from '@fortawesome/free-solid-svg-icons/faLightbulb';
import { faLink } from '@fortawesome/free-solid-svg-icons/faLink';
import { faList } from '@fortawesome/free-solid-svg-icons/faList';
import { faLock } from '@fortawesome/free-solid-svg-icons/faLock';
import { faMapMarker } from '@fortawesome/free-solid-svg-icons/faMapMarker';
import { faMinus } from '@fortawesome/free-solid-svg-icons/faMinus';
import { faPaperPlane } from '@fortawesome/free-solid-svg-icons/faPaperPlane';
import { faPaperclip } from '@fortawesome/free-solid-svg-icons/faPaperclip';
import { faParagraph } from '@fortawesome/free-solid-svg-icons/faParagraph';
import { faPassport } from '@fortawesome/free-solid-svg-icons/faPassport';
import { faPencilAlt } from '@fortawesome/free-solid-svg-icons/faPencilAlt';
import { faPlay } from '@fortawesome/free-solid-svg-icons/faPlay';
import { faPlus } from '@fortawesome/free-solid-svg-icons/faPlus';
import { faPowerOff } from '@fortawesome/free-solid-svg-icons/faPowerOff';
import { faQuestionCircle } from '@fortawesome/free-solid-svg-icons/faQuestionCircle';
import { faQuoteLeft } from '@fortawesome/free-solid-svg-icons/faQuoteLeft';
import { faQuoteRight } from '@fortawesome/free-solid-svg-icons/faQuoteRight';
import { faSave } from '@fortawesome/free-solid-svg-icons/faSave';
import { faSearch } from '@fortawesome/free-solid-svg-icons/faSearch';
import { faServer } from '@fortawesome/free-solid-svg-icons/faServer';
import { faSearchMinus } from '@fortawesome/free-solid-svg-icons/faSearchMinus';
import { faSearchPlus } from '@fortawesome/free-solid-svg-icons/faSearchPlus';
import { faSitemap } from '@fortawesome/free-solid-svg-icons/faSitemap';
import { faSort } from '@fortawesome/free-solid-svg-icons/faSort';
import { faSortAlphaDown } from '@fortawesome/free-solid-svg-icons/faSortAlphaDown';
import { faSpinner } from '@fortawesome/free-solid-svg-icons/faSpinner';
import { faSquare } from '@fortawesome/free-regular-svg-icons/faSquare';
import { faStar } from '@fortawesome/free-solid-svg-icons/faStar';
import { faStop } from '@fortawesome/free-solid-svg-icons/faStop';
import { faSync } from '@fortawesome/free-solid-svg-icons/faSync';
import { faTag } from '@fortawesome/free-solid-svg-icons/faTag';
import { faTerminal } from '@fortawesome/free-solid-svg-icons/faTerminal';
import { faTh } from '@fortawesome/free-solid-svg-icons/faTh';
import { faTimes } from '@fortawesome/free-solid-svg-icons/faTimes';
import { faTimesCircle } from '@fortawesome/free-solid-svg-icons/faTimesCircle';
import { faToggleOff } from '@fortawesome/free-solid-svg-icons/faToggleOff';
import { faToggleOn } from '@fortawesome/free-solid-svg-icons/faToggleOn';
import { faTrashAlt } from '@fortawesome/free-solid-svg-icons/faTrashAlt';
import { faUndo } from '@fortawesome/free-solid-svg-icons/faUndo';
import { faUpload } from '@fortawesome/free-solid-svg-icons/faUpload';
import { faVideo } from '@fortawesome/free-solid-svg-icons/faVideo';
import { faEye } from '@fortawesome/free-solid-svg-icons/faEye';
import { faArrowsAlt } from '@fortawesome/free-solid-svg-icons/faArrowsAlt';
import { faEyeSlash } from '@fortawesome/free-solid-svg-icons/faEyeSlash';
import { faUsers } from '@fortawesome/free-solid-svg-icons/faUsers';
import { faUserTimes } from '@fortawesome/free-solid-svg-icons/faUserTimes';
import { faHandPaper } from '@fortawesome/free-solid-svg-icons/faHandPaper';
import { faExternalLinkAlt } from '@fortawesome/free-solid-svg-icons/faExternalLinkAlt';
import { faTasks } from '@fortawesome/free-solid-svg-icons/faTasks';
import { saveAndNext } from './save-and-next';
import { exportCsv } from './export-csv';
import { copyFrom } from './copy-from';
import { funnelFilter } from './funnel-filter';

const icons = {
  faAlignJustify,
  faAngleLeft,
  faAngleRight,
  faArrowLeft,
  faArrowsAlt,
  faArrowUp,
  faArrowRight,
  faBars,
  faBook,
  faBullhorn,
  faCalculator,
  faCalendar,
  faCaretDown,
  faCaretUp,
  faChartBar,
  faCheck,
  faCheckCircle,
  faChevronLeft,
  faChevronRight,
  faCircle,
  faClock,
  faClone,
  faCloudUploadAlt,
  faCloudDownloadAlt,
  faCode,
  faCoffee,
  faCog,
  faComment,
  faCubes,
  faEmptyCircle,
  faEnvelope,
  faExchangeAlt,
  faExclamationTriangle,
  faFlask,
  faFile,
  faFilter,
  faFilePdf,
  faFileImage,
  faFont,
  faGavel,
  faHome,
  faImage,
  faInfoCircle,
  faLanguage,
  faLightbulb,
  faLink,
  faList,
  faLock,
  faMapMarker,
  faMinus,
  faPaperPlane,
  faPaperclip,
  faParagraph,
  faPassport,
  faPencilAlt,
  faPlay,
  faPlus,
  faPowerOff,
  faQuestionCircle,
  faQuoteLeft,
  faQuoteRight,
  faSave,
  faSearch,
  faServer,
  faSearchMinus,
  faSearchPlus,
  faSitemap,
  faSort,
  faSortAlphaDown,
  faSpinner,
  faSquare,
  faStar,
  faStop,
  faSync,
  faTag,
  faTerminal,
  faTh,
  faTimes,
  faTimesCircle,
  faToggleOff,
  faToggleOn,
  faTrashAlt,
  faUndo,
  faUpload,
  faVideo,
  faEye,
  faEyeSlash,
  faUsers,
  faUserTimes,
  faHandPaper,
  faExternalLinkAlt,
  saveAndNext,
  faTasks,
  exportCsv,
  copyFrom,
  funnelFilter,
};

export const loadIcons = () => {
  library.add(...Object.keys(icons).map(key => icons[key]));
};

export const iconNames = Object.keys(icons).map(key => icons[key].iconName);
