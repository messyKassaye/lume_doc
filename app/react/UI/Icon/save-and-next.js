/**
 *
 *
 * @format
 */

/* eslint-disable max-len */

/**
 * Original svg code for the save and next icon
 * <svg width="448px" height="448px" viewBox="0 0 448 448" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
 *  <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
 *      <path d="M316.118,0 C328.646297,0 340.669588,4.89796549 349.627934,13.6333306 L350.059,14.059 L433.941,97.941 C442.799876,106.799811 447.838459,118.764857 447.996182,131.276197 L448,131.882 L448,400 C448,426.2449 426.937651,447.569698 400.793778,447.993569 L400,448 L48,448 C21.7551,448 0.430302,426.937651 0.00643053,400.793778 L0,400 L0,48 C0,21.7551 21.062349,0.430302 47.2062225,0.00643053 L48,0 L316.118,0 Z M229.58855,116.6485 C224.81453,111.783833 217.011829,111.783833 212.186475,116.6485 L212.186475,116.6485 L200.790425,128.137394 C195.862404,133.053813 195.965071,141.075338 200.995759,145.888253 L200.995759,145.888253 L262.852695,205.29929 L115.320054,205.29929 C108.492691,205.29929 103,210.83673 103,217.719716 L103,217.719716 L103,234.280284 C103,241.16327 108.492691,246.70071 115.320054,246.70071 L115.320054,246.70071 L262.852695,246.70071 L200.995759,306.111747 C196.016404,310.924662 195.913737,318.946187 200.790425,323.862606 L200.790425,323.862606 L212.186475,335.3515 C216.960495,340.216167 224.763196,340.216167 229.58855,335.3515 L229.58855,335.3515 L329.380984,234.74605 C334.206339,229.933135 334.206339,222.066865 329.380984,217.202198 L329.380984,217.202198 Z" id="Combined-Shape"></path>
 *  </g>
 * </svg>
 */

const saveAndNextSvgPath =
  'M316.118,0 C328.646297,0 340.669588,4.89796549 349.627934,13.6333306 L350.059,14.059 L433.941,97.941 C442.799876,106.799811 447.838459,118.764857 447.996182,131.276197 L448,131.882 L448,400 C448,426.2449 426.937651,447.569698 400.793778,447.993569 L400,448 L48,448 C21.7551,448 0.430302,426.937651 0.00643053,400.793778 L0,400 L0,48 C0,21.7551 21.062349,0.430302 47.2062225,0.00643053 L48,0 L316.118,0 Z M229.58855,116.6485 C224.81453,111.783833 217.011829,111.783833 212.186475,116.6485 L212.186475,116.6485 L200.790425,128.137394 C195.862404,133.053813 195.965071,141.075338 200.995759,145.888253 L200.995759,145.888253 L262.852695,205.29929 L115.320054,205.29929 C108.492691,205.29929 103,210.83673 103,217.719716 L103,217.719716 L103,234.280284 C103,241.16327 108.492691,246.70071 115.320054,246.70071 L115.320054,246.70071 L262.852695,246.70071 L200.995759,306.111747 C196.016404,310.924662 195.913737,318.946187 200.790425,323.862606 L200.790425,323.862606 L212.186475,335.3515 C216.960495,340.216167 224.763196,340.216167 229.58855,335.3515 L229.58855,335.3515 L329.380984,234.74605 C334.206339,229.933135 334.206339,222.066865 329.380984,217.202198 L329.380984,217.202198 Z';
export const saveAndNext = {
  prefix: 'fas',
  iconName: 'save-and-next',
  icon: [448, 448, [], 'e001', saveAndNextSvgPath],
};
