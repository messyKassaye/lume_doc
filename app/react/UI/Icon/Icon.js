/** @format */

import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { allLanguages as languagesList } from 'shared/languagesList';
import { loadIcons } from './library';

loadIcons();

const Icon = ({ locale, directionAware, ...ownProps }) => {
  const languageData = languagesList.find(l => l.key === locale);
  return (
    <FontAwesomeIcon {...ownProps} flip={languageData && languageData.rtl ? 'horizontal' : null} />
  );
};

Icon.defaultProps = {
  locale: '',
  directionAware: false,
};

Icon.propTypes = {
  locale: PropTypes.string,
  directionAware: PropTypes.bool,
};

export const mapStateToProps = ({ locale }) => ({ locale });

export default connect(mapStateToProps, () => ({}))(Icon);
