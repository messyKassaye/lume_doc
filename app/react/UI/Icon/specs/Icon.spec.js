import React from 'react';
import { shallow } from 'enzyme';

import Icon, { mapStateToProps } from '../Icon';

describe('Icon', () => {
  let component;
  let props;

  const render = () => {
    component = shallow(<Icon.WrappedComponent {...props} />);
  };

  it('should instantiate a FontAwesomeIcon', () => {
    props = { icon: 'angle-right', size: 'xs' };
    render();
    expect(component).toMatchSnapshot();
  });

  it('should allow configuring the icon as directionAware', () => {
    props = { icon: 'angle-left', directionAware: true, locale: 'es' };
    render();
    expect(component).toMatchSnapshot();

    props = { icon: 'angle-left', directionAware: true, locale: 'ar' };
    render();
    expect(component).toMatchSnapshot();
  });

  describe('MapStateToProps', () => {
    it('should map the locale', () => {
      const state = { locale: 'en' };

      const mappedProps = mapStateToProps(state);
      expect(mappedProps.locale).toBe(state.locale);
    });
  });
});
