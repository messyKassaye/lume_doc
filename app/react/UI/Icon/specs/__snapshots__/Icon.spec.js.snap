// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should allow configuring the icon as directionAware 1`] = `
<FontAwesomeIcon
  border={false}
  className=""
  fixedWidth={false}
  flip={null}
  icon="angle-left"
  inverse={false}
  listItem={false}
  mask={null}
  pull={null}
  pulse={false}
  rotation={null}
  size={null}
  spin={false}
  swapOpacity={false}
  symbol={false}
  title=""
  transform={null}
/>
`;

exports[`Icon should allow configuring the icon as directionAware 2`] = `
<FontAwesomeIcon
  border={false}
  className=""
  fixedWidth={false}
  flip="horizontal"
  icon="angle-left"
  inverse={false}
  listItem={false}
  mask={null}
  pull={null}
  pulse={false}
  rotation={null}
  size={null}
  spin={false}
  swapOpacity={false}
  symbol={false}
  title=""
  transform={null}
/>
`;

exports[`Icon should instantiate a FontAwesomeIcon 1`] = `
<FontAwesomeIcon
  border={false}
  className=""
  fixedWidth={false}
  flip={null}
  icon="angle-right"
  inverse={false}
  listItem={false}
  mask={null}
  pull={null}
  pulse={false}
  rotation={null}
  size="xs"
  spin={false}
  swapOpacity={false}
  symbol={false}
  title=""
  transform={null}
/>
`;
